### PR TITLE
While Loops: handle exiting a loop from within a conditional block

### PR DESCRIPTION
**Pull request by nbdaaron**
The existing logic doesn't recognize that you can break out of loops inside of an `if` statement, since it only checks for `LEAVES_LOOP` in the main scope.

### CODE REVIEW

File: pyanalyze/stacked_scopes.py [hunk 1]
```diff
@@ -1144,7 +1144,7 @@ def loop_scope(self) -> Iterator[SubScope]:
         with self.subscope() as main_scope:
             loop_scopes.append(main_scope)
             with qcore.override(self, "current_loop_scopes", loop_scopes):
-                yield main_scope
+                yield loop_scopes
```
Comment (JelleZijlstra) [inline pyanalyze/stacked_scopes.py:1147]: The annotation on line 1142 is wrong now, it should be `Iterable[Sequence[Subscope]]` I think.

For some reason this failed the self check only on 3.6.

File: pyanalyze/test_name_check_visitor.py [hunk 2]
```diff
@@ -1663,6 +1663,21 @@ def capybara(condition):
...
+class TestNestedLoop(TestNameCheckVisitorBase):
+    @assert_passes()
+    def test(self):
+        def capybara(x: int):
+            v = 1
+            while x < 2:
+                while True:
+                    if x == 0:
+                        assert_is_value(
+                            v, MultiValuedValue([KnownValue(1), KnownValue(2)])
```
Comment (JelleZijlstra) [inline pyanalyze/test_name_check_visitor.py:1675]:
```suggestion
                            v, KnownValue(1) | KnownValue(2)
```
More concise syntax (probably will also need to reformat).

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Thanks, good catch!

This should also get an entry in `docs/changelog.md`.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
The self-check is still failing on 3.6 because the `StackedScopes.loop_scope` annotation will also need to be changed.

I'm going to look into why we don't catch this on 3.7+.

**Comment by nbdaaron** [comment]
I can update that. does the type for the base Scope class not need to be changed?

**Comment by JelleZijlstra** [comment]
> does the type for the base Scope class not need to be changed

Yes I think so. That's another thing that pyanalyze should catch by itself.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**